pandas/parser.pyx

```diff
@@ -1468,6 +1505,97 @@ cdef _string_box_decode(parser_t *parser, int col,
 
     return result, na_count
 
+@cython.boundscheck(False)
+cdef _categorical_convert(parser_t *parser, int col,
```

Comment:
ahh, yes that makes sense. Actually the same strategy for caching in date parsing. Ok, but instead of rewriting the factorization, can you simply call it with the strings? much simpler